source/MapPanel.cpp

```diff
@@ -650,30 +650,24 @@ void MapPanel::Select(const System *system)
 	}
 	else if(shift)
 	{
-		DistanceMap localDistance(player, plan.front());
-		if(localDistance.Days(system) <= 0)
+		const System *planEnd = plan.front();
+		if(system == planEnd)
 			return;
 
+		DistanceMap localDistance(player, planEnd, system);
+		if(!localDistance.HasRoute(system))
```

Comment:
HasRoute defaults to use the destination (i.e. `system`) if you don't provide it. So you can simply omit it.